llvm/test/CodeGen/RISCV/rvv/compressstore.ll

```diff
@@ -0,0 +1,17545 @@
+; RV32-NEXT:    mv a0, a5
+; RV32-NEXT:    andi a4, a3, 512
+; RV32-NEXT:    bnez a4, .LBB8_435
+; RV32-NEXT:    j .LBB8_188
+; RV32-NEXT:  .LBB8_435: # %cond.store505
+; RV32-NEXT:    li a4, 128
+; RV32-NEXT:    lui a5, 3
+; RV32-NEXT:    addi a5, a5, -1152
+; RV32-NEXT:    add a5, sp, a5
+; RV32-NEXT:    vsetvli zero, a4, e8, m8, ta, ma
+; RV32-NEXT:    vse8.v v24, (a5)
+; RV32-NEXT:    lbu a4, 508(a2)
+; RV32-NEXT:    addi a5, a0, 1
+; RV32-NEXT:    sb a4, 0(a0)
+; RV32-NEXT:    mv a0, a5
+; RV32-NEXT:    andi a4, a3, 1024
+; RV32-NEXT:    bnez a4, .LBB8_436
+; RV32-NEXT:    j .LBB8_189
+; RV32-NEXT:  .LBB8_436: # %cond.store508
+; RV32-NEXT:    li a4, 128
+; RV32-NEXT:    lui a5, 3
+; RV32-NEXT:    addi a5, a5, -1280
+; RV32-NEXT:    add a5, sp, a5
+; RV32-NEXT:    vsetvli zero, a4, e8, m8, ta, ma
+; RV32-NEXT:    vse8.v v24, (a5)
+; RV32-NEXT:    lbu a4, 381(a2)
+; RV32-NEXT:    addi a5, a0, 1
+; RV32-NEXT:    sb a4, 0(a0)
+; RV32-NEXT:    mv a0, a5
+; RV32-NEXT:    slli a4, a3, 20
+; RV32-NEXT:    bltz a4, .LBB8_437
+; RV32-NEXT:    j .LBB8_190
+; RV32-NEXT:  .LBB8_437: # %cond.store511
+; RV32-NEXT:    li a4, 128
+; RV32-NEXT:    lui a5, 3
+; RV32-NEXT:    addi a5, a5, -1408
+; RV32-NEXT:    add a5, sp, a5
+; RV32-NEXT:    vsetvli zero, a4, e8, m8, ta, ma
+; RV32-NEXT:    vse8.v v24, (a5)
+; RV32-NEXT:    lbu a4, 254(a2)
+; RV32-NEXT:    addi a5, a0, 1
+; RV32-NEXT:    sb a4, 0(a0)
+; RV32-NEXT:    mv a0, a5
+; RV32-NEXT:    slli a4, a3, 19
+; RV32-NEXT:    bltz a4, .LBB8_438
+; RV32-NEXT:    j .LBB8_191
+; RV32-NEXT:  .LBB8_438: # %cond.store514
+; RV32-NEXT:    li a4, 128
+; RV32-NEXT:    li a5, 21
+; RV32-NEXT:    slli a5, a5, 9
+; RV32-NEXT:    add a5, sp, a5
+; RV32-NEXT:    vsetvli zero, a4, e8, m8, ta, ma
+; RV32-NEXT:    vse8.v v24, (a5)
+; RV32-NEXT:    lbu a4, 127(a2)
+; RV32-NEXT:    addi a5, a0, 1
+; RV32-NEXT:    sb a4, 0(a0)
+; RV32-NEXT:    mv a0, a5
+; RV32-NEXT:    slli a4, a3, 18
+; RV32-NEXT:    bgez a4, .LBB8_522
+; RV32-NEXT:    j .LBB8_192
+; RV32-NEXT:  .LBB8_522: # %cond.store514
+; RV32-NEXT:    j .LBB8_193
+; RV32-NEXT:  .LBB8_439: # %cond.store520
+; RV32-NEXT:    li a4, 128
+; RV32-NEXT:    lui a5, 3
+; RV32-NEXT:    addi a5, a5, -1792
+; RV32-NEXT:    add a5, sp, a5
+; RV32-NEXT:    vsetvli zero, a4, e8, m8, ta, ma
+; RV32-NEXT:    vse8.v v24, (a5)
+; RV32-NEXT:    lbu a4, 2032(a2)
+; RV32-NEXT:    addi a5, a0, 1
+; RV32-NEXT:    sb a4, 0(a0)
+; RV32-NEXT:    mv a0, a5
+; RV32-NEXT:    slli a4, a3, 16
+; RV32-NEXT:    bltz a4, .LBB8_440
+; RV32-NEXT:    j .LBB8_195
+; RV32-NEXT:  .LBB8_440: # %cond.store523
+; RV32-NEXT:    li a4, 128
+; RV32-NEXT:    lui a5, 3
+; RV32-NEXT:    addi a5, a5, -1920
+; RV32-NEXT:    add a5, sp, a5
+; RV32-NEXT:    vsetvli zero, a4, e8, m8, ta, ma
+; RV32-NEXT:    vse8.v v24, (a5)
+; RV32-NEXT:    lbu a4, 1905(a2)
+; RV32-NEXT:    addi a5, a0, 1
+; RV32-NEXT:    sb a4, 0(a0)
+; RV32-NEXT:    mv a0, a5
+; RV32-NEXT:    slli a4, a3, 15
+; RV32-NEXT:    bltz a4, .LBB
```

Comment:
Is this codegen right? It's so huge.

Reply:
Looks like there's no alignment information so the scalar stores got split